source/Server/Integration/JiraRestClient.cs

```diff
@@ -114,6 +147,18 @@ HttpClient CreateHttpClient(IOctopusHttpClientFactory octopusHttpClientFactory)
             client.DefaultRequestHeaders.Authorization = authorizationHeader;
             return client;
         }
+
+        public void Dispose()
+        {
+            httpClient.Dispose();
+        }
+    }
+
+    class JiraSearchResponse
```

Comment:
I typically prefer to set this type of class up so that IsSuccess has a private setter. Then you have 2 ctors

```
JiraSearchResponse (string errorMessage)
{
    IsSuccess = false;
    ErrorMessage = errorMessage;
}

JiraSearchResponse (JiraSearchResult result)
{
    IsSuccess = true;
    Result = result;
}
```
you cannot construct the object in the wrong state then.

Alteratively, you could use subtypes for [success](https://github.com/OctopusDeploy/ServerExtensibility/blob/main/source/Server.Extensibility/Results/ResultFromExtension.cs) and [failure](https://github.com/OctopusDeploy/ServerExtensibility/blob/main/source/Server.Extensibility/Results/FailureResultFromExtension.cs) and then not have confusion around when message and result may be null, because they only exist on the object type where they are valid.
The types linked there are actually used in this extension, see [this code below](https://github.com/OctopusDeploy/JiraIntegration/pull/58/files#diff-8528eb959e94329f3c1aceddcd6fcfc78c6fe78c945c24d36b64b4bf39bc3ef7R50)

Reply:
Fixed